packages/playwright/src/worker/testTracing.ts

```diff
@@ -130,10 +130,9 @@ export class TestTracing {
     return `${this._testInfo.testId}${retrySuffix}${ordinalSuffix}`;
   }
 
-  generateNextTraceRecordingPath() {
+  generateNextTraceRecordingPath(): { file: string, commit(): void } {
     const file = path.join(this._artifactsDir, createGuid() + '.zip');
-    this._temporaryTraceFiles.push(file);
-    return file;
+    return { file, commit: () => this._temporaryTraceFiles.push(file) };
```

Comment:
I believe we should also ensure that `stopIfNeeded` waits for all created chunks to be written. With current approach it some of them may not be in _temporaryTraceFiles list yet when stopIfNeeded merges the files.